index.bs

```diff
@@ -486,6 +486,36 @@ reasonable future change. We could run [=process a network cookie access for bou
 in the <a spec="fetch">HTTP-network-or-cache fetch</a> algorithm after step 8.21.1.2,
 "... [=append=] (`Cookie`, cookies) to httpRequest’s [=header list=]. ..."
 
+<h5 id="bounce-tracking-mitigations-storage-access-monkey-patch">Storage Access Monkey Patch</h5>
+
+Each [=top-level traversable=] maintains a record of which sites have accessed storage in the current [=extended navigation=].
+
+Insert the following steps in the <a spec="storage">obtain a storage key</a> algorithm before step 4, "Return key":
+
+1. Run [=process a storage access for bounce tracking mitigations=] given <var ignore>environment</var>.
+
+Issue: TODO: Verify that this algorithm is the correct location to catch precisely those sites which access non-cookie storage.
+
+<div algorithm>
+
+To <dfn>process a storage access for bounce tracking mitigations</dfn>
+given an [=environment=] |environment|, perform the following steps:
```

Comment:
So, the reason `[=obtain a storage key=]` takes environment instead of environment settings is because it can used by service worker during a navigation before the document is created.  See step 14.5 here:

https://w3c.github.io/ServiceWorker/#on-fetch-request-algorithm

With that context, these changes would probably result in us treating every navigation as stateful.  What we really want is to only treat navigations where there is a matching service worker with a fetch event handler as stateful.

So maybe we should make this general hook bail out if it gets passed an environment that is not an environment settings object.  Then we can have a separate monkey patch for the service worker path.

WDYT?

Reply:
Also, note service workers is bad in that it does not use `[=obtain a bottle map=]`.  See w3c/ServiceWorker#1230.